File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        int i = 0;
        int len = nums.length;
        int j = len - 1;
        int p = len - 1;
        int[] res = new int[len];
        while (i <= j) {
            if (Math.abs(nums[i]) < Math.abs(nums[j])) {
                res[p] = nums[j] * nums[j];
                j--;
                p--;
            } else {
                res[p] = nums[i] * nums[i];
                i++;
                p--;
            }
        }
        return res;
    }
}

File: 1019-squares-of-a-sorted-array/squares-of-a-sorted-array.ts
function sortedSquares(nums: number[]): number[] {
    const result: number[] = new Array(nums.length);
    let insertIdx = nums.length - 1;
    let leftPointer = 0;
    let rightPointer = nums.length - 1;
    while (leftPointer <= rightPointer) {
        const a = Math.abs(nums[leftPointer]);
        const b = Math.abs(nums[rightPointer]);
        if (a >= b) {
            result[insertIdx--] = Math.pow(a, 2);
            leftPointer++;
        } else {
            result[insertIdx--] = Math.pow(b, 2);
            rightPointer--;
        }
    }
    return result;
};